File: DSA/Searching/LinearSearch/ls.cpp
//simple linear search implementation in cpp
#include <iostream>
using namespace std;

class solution {
public:
    int linearSearch(int arr[], int n, int key) { //declares the function linearSearch with parameters arr (array), n (size of array), key (element to search)
        for (int i = 0; i < n; i++) {
            if (arr[i] == key) {
                return i; //key found at index i
            }
        }
        return -1; //key not found
    }
};

File: practice/Typescript/search/ls.ts
//linear search in typescript

function linearSearch<T>(arr: T[], target: T): number {
    for (let i = 0; i < arr.length; i++) {
        if (arr[i] === target) {
            return i; // Return the index if the target is found
        }
    }
    return -1; // Return -1 if the target is not found
}

// Example usage:
const numbers = [10, 20, 30, 40, 50];
const targetNumber = 30;
const resultIndex = linearSearch(numbers, targetNumber);
console.log(`Element found at index: ${resultIndex}`); // Output: Element found at index: 2

const strings = ["apple", "banana", "cherry", "date"];
const targetString = "cherry";
const resultStringIndex = linearSearch(strings, targetString);
console.log(`Element found at index: ${resultStringIndex}`); // Output: Element found at index: 2